packages/wonder-blocks-switch/src/components/switch.tsx

```diff
@@ -0,0 +1,240 @@
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+
+import {
+    AriaProps,
+    UniqueIDProvider,
+    View,
+    addStyle,
+} from "@khanacademy/wonder-blocks-core";
+import Color, {mix} from "@khanacademy/wonder-blocks-color";
+import Icon from "@khanacademy/wonder-blocks-icon";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+
+type Props = Pick<
+    AriaProps,
+    "aria-labelledby" | "aria-label" | "aria-describedby"
+> & {
+    /**
+     * Whether this compoonent is checked.
+     */
+    checked: boolean;
+    /**
+     * Whether the switch is disabled. Defaults to `false`.
+     */
+    disabled?: boolean;
+    /**
+     * Optional icon to display on the slider.
+     */
+    icon?: React.ReactElement<typeof Icon>;
+    /**
+     * The unique identifier for the switch.
+     */
+    id?: string;
+    /**
+     * Function to call when the switch is clicked.
+     * @param newCheckedValue
+     * @returns {unknown}
+     */
+    onChange?: (newCheckedState: boolean) => unknown;
+    /**
+     * Test ID used for e2e testing.
+     */
+    testId?: string;
+};
+
+const StyledSpan = addStyle("span");
+const StyledInput = addStyle("input");
+
+const Switch = React.forwardRef(function Switch(
+    props: Props,
+    ref: React.ForwardedRef<HTMLInputElement>,
+) {
+    const {
+        "aria-label": ariaLabel,
+        "aria-labelledby": ariaLabelledBy,
+        "aria-describedby": ariaDescribedBy,
+        checked,
+        disabled = false,
+        icon,
+        id,
+        onChange,
+        testId,
+    } = props;
+
+    const handleClick = () => {
+        if (!disabled && onChange) {
+            onChange(!checked);
+        }
+    };
+
+    const stateStyles = _generateStyles(
+        checked,
+        disabled,
+        onChange !== undefined,
+    );
+
+    let styledIcon: React.ReactElement<typeof Icon> | undefined;
+    if (icon) {
+        styledIcon = React.cloneElement(icon, {
+            size: "small",
+            style: [sharedStyles.icon, stateStyles.icon],
+            ...icon.props,
+        } as Partial<React.ComponentProps<typeof Icon>>);
+    }
+
+    return (
+        <UniqueIDProvider mockOnFirstRender={true} scope="switch">
+            {(ids) => {
+                const uniqueId = id || ids.get("switch");
+
+                return (
+                    <View
+                        onClick={handleClick}
+                        style={[
+                            sharedStyles.switch,
+                            stateStyles.switch,
+                            disabled && sharedStyles.disabled,
+                        ]}
+                        testId={testId}
+                    >
+                        <StyledInput
+                            aria-checked={checked}
```

Comment:
**question:** Giving a second thought on this, maybe it's better to use `checked` directly as this is an input[checkbox] element? wdyt? (cc @nishasy)

Reply:
good point! it seems since it's a native checkbox using `checked` is fine.
source: https://stackoverflow.com/questions/49179545/do-i-need-aria-checked-on-an-actual-checkbox#:~:text=The%20aria%20attributes%20are%20for,and%20other%20attributes%20as%20well).